handlers/aws/utils.py

```diff
@@ -125,13 +125,87 @@ def wrapper(lambda_event: dict[str, Any], lambda_context: context_.Context) -> s
             if apm_client:
                 apm_client.capture_exception()
 
-            shared_logger.exception("exception raised", exc_info=e)
+            shared_logger.exception(
+                "exception raised",
+                exc_info=e,
+                extra={
+                    "summary": summarize_lambda_event(lambda_event, max_records=20),
+                },
+            )
 
             return f"exception raised: {e.__repr__()}"
 
     return wrapper
 
 
+def summarize_lambda_event(event: dict[str, Any], max_records: int = 10) -> dict[str, Any]:
+    """
+    Summarize the lambda event to include only the most relevant information.
+    """
+    summary: dict[str, Any] = {}
+
+    try:
+        records = event.get("Records", [])
+
+        for record in records:
+            event_source = record.get("eventSource", "unknown")
+
+            if event_source == "aws:sqs":
+                aws_sqs_summary = summary.get(
+                    "aws:sqs",
+                    # if the key does not exist, we initialize the summary
+                    {
+                        "total_records": 0,
+                        "records": [],
+                    },
+                )
+
+                # The body contains the SQS message payload which is the S3
+                # notification event encoded as a JSON string.
+                event = json_parser(record["body"])
+
+                # So users know if we included only a
+                # subset of the records.
+                aws_sqs_summary["total_records"] += len(event["Records"])
```

Comment:
I don't understand this very well. When would this happen? So if the summary does not exist, then you initialized it at 0, but if it exists, wouldn't this variable already be correct?

And if not, shouldn't it be changed inside the `for` cycle right below for when you append a new record?

Reply:
Ok, I think I have just realized exactly what you are doing. So `total_records` is the total records there, but `records` is only up to the maximum (so 20).

Reply:
Yeah, I don't want to dump all the records from the lambda event into the log, I don't know how many of them we could receive. 

The lambda can receive up to 10 SQS messages and each message can contain multiple notifications. I want to protect and cap the log message size.

In the resulting log event:

```yaml
    "summary": {
        "aws:sqs": {
            "records": [ ... ],
            "total_records": 1
        }
```

- `records` contains the actual records
- `total_records` contains the total number of records in the lambda event

I don't feel this as confusing, but something like `first_{max_records}_records` is much better (I am using `first` because the summary collects the first max_records records in the lambda event).

Reply:
If we only have 1 notification the end result is:

```json
  "summary": {
        "aws:sqs": {
            "first_20_records": [{"key": "AWSLogs/blah/key"}],
            "total_records": 1
        }
```

Reply:
That looks perfect!

Reply:
Here is a shot from real life:

![image](https://github.com/user-attachments/assets/ecd04da7-736a-43c3-a544-4ecb0aaba038)